openverse_catalog/dags/providers/provider_api_scripts/stocksnap.py

```diff
@@ -25,178 +25,162 @@
-    except (TypeError, KeyError, AttributeError):
+    def get_next_query_params(self, old_query_params, **kwargs):
+        # StockSnap uses /{page} at the end of the endpoint url instead of query params.
+        self._page_counter += 1
         return None
-    foreign_landing_url = f"https://{HOST}/photo/{foreign_id}"
-    image_url, width, height = _get_image_info(media_data)
-    if image_url is None:
-        logger.info("Found no image url.")
-        logger.info(f"{json.dumps(media_data, indent=2)}")
-        return None
-    title = _get_title(media_data)
-    if title is None:
-        logger.info("Found no image title.")
-        logger.info(f"{json.dumps(media_data, indent=2)}")
+
+    def get_media_type(self, record):
+        return "image"
+
+    @property
+    def endpoint(self):
+        return f"{ENDPOINT_BASE}/{self._page_counter}"
+
+    def get_should_continue(self, response_json):
+        return bool(response_json.get("nextPage"))
+
+    def get_batch_data(self, response_json):
+        """
+        Take an API response and return the list of records.
+        """
+        if response_json:
+            return response_json.get("results")
         return None
-    creator, creator_url = _get_creator_data(media_data)
-    metadata = _get_metadata(media_data)
-    tags = _get_tags(media_data)
-
-    return {
-        "title": title,
-        "creator": creator,
-        "creator_url": creator_url,
-        "foreign_identifier": foreign_id,
-        "foreign_landing_url": foreign_landing_url,
-        "image_url": image_url,
-        "filesize": _get_filesize(image_url),
-        "filetype": "jpg",
-        "height": height,
-        "width": width,
-        "license_info": license_info,
-        "meta_data": metadata,
-        "raw_tags": tags,
-        "category": prov.DEFAULT_IMAGE_CATEGORY[PROVIDER],
-    }
 
+    def get_record_data(self, data):
+        """
+        Parse out the necessary information (license info, urls, etc) from the record
+        data (single image dict-like thing) into a dictionary.
+        """
+        try:
+            foreign_id = data["img_id"]
+        except (TypeError, KeyError, AttributeError):
+            return None
+
+        foreign_landing_url = f"https://{HOST}/photo/{foreign_id}"
+
+        image_url, width, height = self._get_image_info(data)
+        if image_url is None:
+            logger.info("Found no image url.")
+            logger.info(f"{json.dumps(data, indent=2)}")
+            return None
+
+        title = self._get_title(data)
+        if title is None:
+            logger.info("Found no image title.")
+            logger.info(f"{json.dumps(data, indent=2)}")
+            return None
+
+        creator, creator_url = self._get_creator_data(data)
+        metadata = self._get_metadata(data)
+        tags = self._get_tags(data)
+        filesize = self._get_filesize(image_url)
+        lic = get_license_info("https://creativecommons.org/publicdomain/zero/1.0/")
```

Comment:
This will be processed for each image so declaring it at the beginning as a class attribute would be slightly more efficient :)

```suggestion
        lic = self.license
```

Reply:
Of course! Sorry, I think I was in inaturalist head where different images have different licenses. Will do!